src/rrt/Tree.hpp

```diff
@@ -52,7 +68,10 @@ class Node {
      */
     const T& state() const { return _state; }
 
+    double* coordinates() { return _vec.data(); }
```

Comment:
What is the function for? If it doesn't need to return a double* its best not to return one since its not immediately obvious what to do with a double*. Like the most obvious is that it points to a single double. but in this case it points to a array start. Hmm Also actually it would be best to have more documentation on the public functions and classes if possible.